yaml/secret.go

```diff
@@ -89,6 +90,29 @@ func (s *SecretSlice) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		if secret.Origin.Empty() && len(secret.Type) == 0 {
 			secret.Type = constants.SecretRepo
 		}
+
+		// implicitly set `pull` field if empty
+		if len(secret.Origin.Pull) == 0 {
+			secret.Origin.Pull = constants.PullNotPresent
+		}
+
+		// TODO: remove this in a future release
+		//
+		// handle true deprecated pull policy
+		//
+		// a `true` pull policy equates to `always`
+		if strings.EqualFold(secret.Origin.Pull, "true") {
+			secret.Origin.Pull = constants.PullAlways
+		}
+
+		// TODO: remove this in a future release
+		//
+		// handle false deprecated pull policy
+		//
+		// a `false` pull policy equates to `not_present`
+		if strings.EqualFold(secret.Origin.Pull, "false") {
+			secret.Origin.Pull = constants.PullNotPresent
+		}
```

Comment:
Set the deprecated `false` pull policy to `not_present` for the `github.com/go-vela/types/yaml.Origin` type